src/devices/device_constructors/hydrogeneration_constructor.jl

```diff
@@ -14,12 +14,55 @@ function construct_device!(canonical::Canonical, sys::PSY.System,
         return
     end
 
-    error("Currently only HydroFixed Formulation is Enabled")
+    #Variables
+    activepower_variables!(canonical, devices);
```

Comment:
We should add here some kind of constraint that has a budget. Something of the type 
```
sum(P_hy[t] for t in time) <= budget
```

Reply:
I'm wrong to assume the budget would be passed as a forecast ? And if so does it have a 24hr timestep  or  hourly ?

Reply:
You need to take into account the resolution to calculate this.